core/src/main/java/io/kestra/core/models/tasks/runners/TaskRunner.java

```diff
@@ -130,6 +131,24 @@ public String toAbsolutePath(RunContext runContext, TaskCommands taskCommands, S
         return windowsToUnixPath(workingDir + "/" + relativePath);
     }
 
+    public List<String> renderCommands(RunContext runContext, TaskCommands taskCommands) throws IllegalVariableEvaluationException, IOException {
+        List<String> renderedCommands = this.renderCommandsFromList(runContext, taskCommands, taskCommands.getCommands());
+        List<String> renderedBeforeCommands = this.renderCommandsFromList(runContext, taskCommands, taskCommands.getBeforeCommands());
+        List<String> renderedInterpreter = this.renderCommandsFromList(runContext, taskCommands, taskCommands.getInterpreter());
+
+        return Stream.of(renderedInterpreter, renderedBeforeCommands, renderedCommands)
+            .flatMap(Collection::stream).toList();
+    }
+
+    public List<String> renderCommandsFromList(RunContext runContext, TaskCommands taskCommands, Property<List<String>> commands) throws IllegalVariableEvaluationException, IOException {
```

Comment:
```suggestion
    private List<String> renderCommandsFromList(RunContext runContext, TaskCommands taskCommands, Property<List<String>> commands) throws IllegalVariableEvaluationException, IOException {
```

It seems it's only used by the `renderCommands` method so better to make it private.

Reply:
Made method `private`